common/src/colors.cpp

```diff
@@ -299,30 +299,297 @@ static const unsigned char GLASBEY_LUT[] =
+  36 , 134, 142,
+  35 , 135, 142,
+  35 , 136, 142,
+  35 , 137, 142,
+  34 , 138, 141,
+  34 , 139, 141,
+  34 , 140, 141,
+  33 , 141, 141,
+  33 , 142, 141,
+  33 , 143, 141,
+  32 , 144, 141,
+  32 , 145, 140,
+  32 , 146, 140,
+  32 , 147, 140,
+  31 , 147, 140,
+  31 , 148, 140,
+  31 , 149, 139,
+  31 , 150, 139,
+  31 , 151, 139,
+  30 , 152, 139,
+  30 , 153, 138,
+  30 , 154, 138,
+  30 , 155, 138,
+  30 , 156, 137,
+  30 , 157, 137,
+  30 , 158, 137,
+  30 , 159, 136,
+  30 , 160, 136,
+  31 , 161, 136,
+  31 , 162, 135,
+  31 , 163, 135,
+  31 , 163, 134,
+  32 , 164, 134,
+  32 , 165, 134,
+  33 , 166, 133,
+  33 , 167, 133,
+  34 , 168, 132,
+  35 , 169, 131,
+  35 , 170, 131,
+  36 , 171, 130,
+  37 , 172, 130,
+  38 , 173, 129,
+  39 , 174, 129,
+  40 , 175, 128,
+  41 , 175, 127,
+  42 , 176, 127,
+  43 , 177, 126,
+  44 , 178, 125,
+  46 , 179, 124,
+  47 , 180, 124,
+  48 , 181, 123,
+  50 , 182, 122,
+  51 , 183, 121,
+  53 , 183, 121,
+  54 , 184, 120,
+  56 , 185, 119,
+  57 , 186, 118,
+  59 , 187, 117,
+  61 , 188, 116,
+  62 , 189, 115,
+  64 , 190, 114,
+  66 , 190, 113,
+  68 , 191, 112,
+  70 , 192, 111,
+  72 , 193, 110,
+  73 , 194, 109,
+  75 , 194, 108,
+  77 , 195, 107,
+  79 , 196, 106,
+  81 , 197, 105,
+  83 , 198, 104,
+  85 , 198, 102,
+  88 , 199, 101,
+  90 , 200, 100,
+  92 , 201, 99 ,
+  94 , 201, 98 ,
+  96 , 202, 96 ,
+  98 , 203, 95 ,
+  101, 204, 94 ,
+  103, 204, 92 ,
+  105, 205, 91 ,
+  108, 206, 90 ,
+  110, 206, 88 ,
+  112, 207, 87 ,
+  115, 208, 85 ,
+  117, 208, 84 ,
+  119, 209, 82 ,
+  122, 210, 81 ,
+  124, 210, 79 ,
+  127, 211, 78 ,
+  129, 212, 76 ,
+  132, 212, 75 ,
+  134, 213, 73 ,
+  137, 213, 72 ,
+  139, 214, 70 ,
+  142, 215, 68 ,
+  144, 215, 67 ,
+  147, 216, 65 ,
+  149, 216, 63 ,
+  152, 217, 62 ,
+  155, 217, 60 ,
+  157, 218, 58 ,
+  160, 218, 57 ,
+  163, 219, 55 ,
+  165, 219, 53 ,
+  168, 220, 51 ,
+  171, 220, 50 ,
+  173, 221, 48 ,
+  176, 221, 46 ,
+  179, 221, 45 ,
+  181, 222, 43 ,
+  184, 222, 41 ,
+  187, 223, 39 ,
+  189, 223, 38 ,
+  192, 223, 36 ,
+  195, 224, 35 ,
+  197, 224, 33 ,
+  200, 225, 32 ,
+  203, 225, 30 ,
+  205, 225, 29 ,
+  208, 226, 28 ,
+  211, 226, 27 ,
+  213, 226, 26 ,
+  216, 227, 25 ,
+  219, 227, 24 ,
+  221, 227, 24 ,
+  224, 228, 24 ,
+  226, 228, 24 ,
+  229, 228, 24 ,
+  232, 229, 25 ,
+  234, 229, 25 ,
+  237, 229, 26 ,
+  239, 230, 27 ,
+  242, 230, 28 ,
+  244, 230, 30 ,
+  247, 230, 31 ,
+  249, 231, 33 ,
+  251, 231, 35 ,
+  254, 231, 36 ,
+};
+
 /// Number of colors in Glasbey lookup table
 static const size_t GLASBEY_LUT_SIZE = sizeof (GLASBEY_LUT) / (sizeof (GLASBEY_LUT[0]) * 3);
 
-pcl::RGB
-pcl::GlasbeyLUT::at (size_t color_id)
+/// Number of colors in Viridis lookup table
+static const unsigned int VIRIDIS_LUT_SIZE = sizeof (VIRIDIS_LUT) / (sizeof (VIRIDIS_LUT[0]) * 3);
+
+static const unsigned char* LUTS[] = { GLASBEY_LUT, VIRIDIS_LUT };
+static const unsigned int LUT_SIZES[] = { GLASBEY_LUT_SIZE, VIRIDIS_LUT_SIZE };
```

Comment:
There's a minor inconsistency with the size types between `VIRIDIS_LUT_SIZE` and `GLASBEY_LUT_SIZE`. I can understand not fully requiring the entire precision of `size_t` but I don't really understand why go for a signed type.

Reply:
Good catch. This commit was fermenting in my cellar since April. In the meantime #2297 landed changing the type of `GLASBEY_LUT_SIZE` to `size_t` and I was not attentive enough when resolving conflicts.